### PR TITLE
Give Thermometer enable/disable methods

### PR DESCRIPTION
**Pull request by islemaster**
Addresses https://github.com/rwaldron/johnny-five/issues/1297 by giving the `Thermometer` component `enable()` and `disable()` methods.

I've added tests but I'm a little confused by the structure of the Thermometer unit tests - this feature lives on the prototype and should be the same for all controllers, but it looks like it has to be tested per-controller?  But similar tests aren't applied to all controllers either.  Let me know if I should make sure these tests get added to every controller, or if we can run them just once on the default controller, etc.

Thanks!

### CODE REVIEW

**Comment by rwaldron** [comment]
> I've added tests but I'm a little confused by the structure of the Thermometer unit tests - this feature lives on the prototype and should be the same for all controllers, but it looks like it has to be tested per-controller? 

This is what I _try_ to do, but not what always ends up happening, for a variety of reasons. 

> But similar tests aren't applied to all controllers either. 

Such is the result of a relaxed contribution policy, but I'm ok with the incurred technical debt (for now).

> Let me know if I should make sure these tests get added to every controller,

Don't worry about that for now :)

> or if we can run them just once on the default controller, etc.

This should be fine for now.

File: lib/thermometer.js [hunk 1]
```diff
@@ -943,7 +941,13 @@ function Thermometer(opts) {
...
   }
 
-  priv.set(this, {});
+  var state = {
+    enabled: typeof opts.enabled === "undefined" ? true : opts.enabled,
+    intervalId: null,
+    freq: opts.freq || 25,
+    previousFreq: opts.freq || 25,
+  };
+  priv.set(this, state);
```
Comment (rwaldron) [inline lib/thermometer.js]: 👍

File: lib/thermometer.js [hunk 2]
```diff
@@ -968,7 +991,22 @@ function Thermometer(opts) {
...
+        state.freq = newFreq;
+        if (state.intervalId) {
+          clearInterval(state.intervalId);
+        }
+
+        if (state.freq !== null) {
+          state.intervalId = setInterval(eventProcessing, newFreq);
+        }
+      }
+    },
```
Comment (rwaldron) [inline lib/thermometer.js]: 👍

File: lib/thermometer.js [hunk 3]
```diff
@@ -983,26 +1021,50 @@ function Thermometer(opts) {
...
-  setInterval(function() {
-    if (raw == null) {
-      return;
-    }
+  // Set the freq property only after the get and set functions are defined
+  // and only if the sensor is not `enabled: false`
+  if (state.enabled) {
+    this.freq = state.freq;
+  }
+}
```
Comment (rwaldron) [inline lib/thermometer.js]: 👍

File: lib/thermometer.js [hunk 4]
```diff
@@ -983,26 +1021,50 @@ function Thermometer(opts) {
...
+  }
 
-    if (this.celsius !== last) {
-      last = this.celsius;
-      this.emit("change", data);
-    }
-  }.bind(this), freq);
-}
+  return this;
+};
```
Comment (rwaldron) [inline lib/thermometer.js]: 👍

File: lib/thermometer.js [hunk 5]
```diff
@@ -983,26 +1021,50 @@ function Thermometer(opts) {
...
+
+  /* istanbul ignore else */
+  if (state.enabled) {
+    state.enabled = false;
+    state.previousFreq = state.freq;
+    this.freq = null;
+  }
+
+  return this;
+};
```
Comment (rwaldron) [inline lib/thermometer.js]: 👍

File: test/thermometer.js [hunk 6]
```diff
@@ -450,7 +547,10 @@ exports["Thermometer -- ANALOG"] = {
       test.expect(1);
       test.equal(digits.fractional(this.thermometer.C), 0);
       test.done();
-    }
+    },
+    enable: testEnable,
+    disable: testDisable,
+    constructDisabled: testConstructDisabled,
```
Comment (rwaldron) [inline test/thermometer.js]: 👍

**Comment by islemaster** [comment]
Done: Documented `enabled`, `freq`, `enable()` and `disable()` on [the Thermometer wiki page](https://github.com/rwaldron/johnny-five/wiki/Thermometer) ([diff](https://github.com/rwaldron/johnny-five/wiki/Thermometer/_compare/215522e4b5ba4c2f25a7a265ded4dcb175737967...246f4c5810249234748a376dd9d6322c8053b524)).  Also documented the same on [the Sensor wiki page](https://github.com/rwaldron/johnny-five/wiki/Sensor) ([diff](https://github.com/rwaldron/johnny-five/wiki/Sensor/_compare/8e6a89add419e42aa89e7ddd4be1940a15d8bb05...d04788c2ff11e9fc001ddedabb3054ad6c2d0513)) which was missing these.

**Pull request merged**